### PR TITLE
chore: adding legacy gov proposal handler for client updates

### PR DESCRIPTION
**Pull request by damiannolan**
## Description

Adding back the legacy v1beta1 gov proposal handler.

ref: #4721 

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#pull-request-targeting)).
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/build/building-modules/11-structure.md) and [Go style guide](../docs/dev/go-style-guide.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/main/testing/README.md#ibc-testing-package).
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`).
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Provide a [commit message](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) to be used for the changelog entry in the PR description for review.
- [ ] Re-reviewed `Files changed` in the Github PR explorer.
- [ ] Review `Codecov Report` in the comment section below once CI passes.

### CODE REVIEW

File: e2e/testsuite/testconfig.go [hunk 1]
```diff
@@ -437,9 +437,8 @@ func getGenesisModificationFunction(cc ChainConfig) func(ibc.ChainConfig, []byte
 // defaultGovv1ModifyGenesis will only modify governance params to ensure the voting period and minimum deposit
 // are functional for e2e testing purposes.
 func defaultGovv1ModifyGenesis(version string) func(ibc.ChainConfig, []byte) ([]byte, error) {
-	var stdlibJSONMarshalling = semverutil.FeatureReleases{MajorVersion: "v8"}
+	stdlibJSONMarshalling := semverutil.FeatureReleases{MajorVersion: "v8"}
```
Comment (damiannolan) [inline e2e/testsuite/testconfig.go:440]: `make format`

File: modules/core/02-client/types/legacy_proposal.go [hunk 2]
```diff
@@ -0,0 +1,60 @@
+package types
+
```
Comment (colin-axner) [inline modules/core/02-client/types/legacy_proposal.go:2]: should we add a comment to say this type is deprecated?

**Review by colin-axner — APPROVED** [review]
Thank you!

File: modules/core/02-client/types/legacy_proposal.go (hunk 2 above)
Comment (damiannolan) [reply modules/core/02-client/types/legacy_proposal.go:2]: Yeah I can add a block comment to this file as well!

**Review by DimitrisJim — APPROVED** [review]
🙏

**Review by charleenfei — APPROVED** [review]
thank you!

**Pull request merged**